src/rust/engine/process_execution/src/docker.rs

```diff
@@ -681,14 +697,86 @@ impl<'a> ContainerCache<'a> {
     Ok(container.id)
   }
 
-  /// Return the container ID of a container running `image` for use as a place to invoke
-  /// build actions as executions within the cached container.
-  pub async fn container_id_for_image(
+  /// Creates a volume for named caches for the given image name. In production usage, the image
+  /// name will have been expanded to include its SHA256 fingerprint, so the named cache will be
+  /// dedicated to a particular image version. That is conservative, so we might consider making
+  /// it configurable whether the image version is attached in future releases.
+  async fn maybe_make_named_cache_volume(
+    docker: &Docker,
+    image: &str,
```

Comment:
This is the image *name*, right? Not the image contents in bytes (which would be huge). Might be worth emphasizing this, since below we hash this string and add it to a name, and you've just mentioned adding the fingerprint of the image contents to the name, so there is potential for reader confusion.

Reply:
Yes, image name.